src/lib/homestore_backend/hs_shard_manager.cpp

```diff
@@ -328,6 +328,22 @@ void HSHomeObject::on_shard_message_commit(int64_t lsn, sisl::blob const& h, hom
             chunk_selector_->select_specific_chunk(blkids.chunk_num());
         }
         if (ctx) { ctx->promise_.setValue(ShardManager::Result< ShardInfo >(shard_info)); }
+
+        // TODO: add shard header blk count to total_occupied_blk_count and handle the case of replaying journal after
```

Comment:
if this is the case, it also involve  in on_blob_put_commit?

Reply:
yes, all the action in `on_commit` will be involved. we need to revisit this later.

Reply:
how other metrics handle this case? IIRC we restart from a checkpoint ?

Reply:
i believe other metrics do not handle this case. as you said,  we definitely restart from a checkpoint. but in homestore, checkpoint is not a real checkpoint, partial persistence might happen. actually, when we do cp_flush, we just start several future to flush those cp of every component. the key point here is that no atomicity of the cp of every component is promised. this is the potential risk of inconsistence。

Reply:
I have post this question in homestore channel. hair will fix it.
https://ebay-eng.slack.com/archives/CAVQXEAKS/p1714099698282209